interface/billing/billing_report.php

```diff
@@ -347,9 +347,8 @@ function set_button_states() {
             const patUrl = 'patient_file/summary/demographics.php?pid=' + encodeURIComponent(pid);
             pid = parseInt(pid, 10);
             enc = parseInt(enc, 10);
+            // Restore the session (getSessionValue calls restoreSession()) and load the encounter
             let curpid = await top.getSessionValue('pid');
-            // Restore the session and load the encounter.
-            top.restoreSession()
```

Comment:
This file isn't the version I merged! This is a previous version using new get session key. However, you were correct removing clearPatient() as not needed because with the awaits functions functions load in correct order and the session is correctly tracked.

Reply:
ok, thanks, must've had an issue when merged master :crying_cat_face:

Reply:
however this is testing well for me, what would you like changed?

Reply:
It appears to work well but if you note that though the correct encounter is shown the encounter history is empty.
Did you see this PR? https://github.com/openemr/openemr/pull/7895

Reply:
yep, merged it https://github.com/openemr/openemr/pull/7896/commits/2addfc04b364c2eb315c239555b3e54603a1b06c

Reply:
Like I said in my testing when clearPatient is used encounter history is empty. Don't know if can be trusted

Reply:
thanks for spotting that

Reply:
![drawing_board](https://github.com/user-attachments/assets/e73ea9df-9c1d-4030-afb1-c10de9569656)

Reply:
It should be working fine with what is in master now.

Reply:
it wasn't updating the medical record dashboard view like in the below image
![Screenshot from 2024-12-30 13-51-27](https://github.com/user-attachments/assets/e5143011-24d8-4cda-9eb0-cb2c473cb4d0)

think have it fixed up now

Reply:
nope, still something up with the dashboard :laughing: